### DISCUSSION TITLE
In colab Link Prediction on MovieLens

### DISCUSSION THREAD

**Discussion by primising** [pyg-team/pytorch_geometric#7879] [Q&A]
In https://colab.research.google.com/drive/1xpzn1Nvai1ygd_P5Yambc_oe4VBPK_ZT?usp=sharing#scrollTo=JMGYv83WzSRr this colab，Heterogeneous Graph Creation，this out:

```
HeteroData(
  user={ node_id=[610] },
  movie={
    node_id=[9742],
    x=[9742, 20]
  },
  (user, rates, movie)={ edge_index=[2, 100836] }
)
```

but in the code:
```python
assert data.edge_types == [("user", "rates", "movie"),
                           ("movie", "rev_rates", "user")].
```
And I can't fill in the correct parameters in the following code,This confused me，I hope someone can help me ,thank you.

**Comment by akihironitta** [comment]
Hi @primising, have you had a chance to look at `T.ToUndirected()` as shown as a hint in the notebook?

```python
# We also need to make sure to add the reverse edges from movies to users
# in order to let a GNN be able to pass messages in both directions.
# We can leverage the `T.ToUndirected()` transform for this from PyG:
```

Here're some references you might find useful:
- https://pytorch-geometric.readthedocs.io/en/2.3.1/notes/heterogeneous.html#heterogeneous-graph-transformations
- https://pytorch-geometric.readthedocs.io/en/2.3.1/generated/torch_geometric.transforms.ToUndirected.html#torch_geometric.transforms.ToUndirected
